### PR TITLE
Fix clang depscand that never returns on Linux

### PR DESCRIPTION
**Pull request by cachemeifyoucan**

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci test linux platform

**Comment by cachemeifyoucan** [comment]
@swift-ci test linux platform

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

File: clang/tools/driver/cc1depscan_main.cpp [hunk 1]
```diff
@@ -1096,7 +1096,9 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
     ShutDown.store(true);
   }
 
-  return 0;
+  // Exit instead of return. Otherwise, it will wait on ~ThreadPool() which will
+  // never return since all threads might still be sleeping on ::accept().
+  ::exit(0);
```
Comment (benlangmuir) [inline clang/tools/driver/cc1depscan_main.cpp:1101]: Are we sure it's safe to `exit` here if there are threads that haven't sync'd? We're going to run global destructors, etc. that could be accessed by the thread.

Comment (cachemeifyoucan) [reply clang/tools/driver/cc1depscan_main.cpp:1101]: To get out of the loop above, you need to hit `NumRunning` is 0, which means all threads are sitting on `accept()`. Then `ShutDown.store(true)` should prevent threads spin up a new job.

So technically, yes, thread should either wait on accept or just returned. Will it crash in all edge cases? I don't know. We need concurrency to save us.

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**